src/provider/systemdmanager/service.rs

```diff
@@ -0,0 +1,321 @@
+            .collect();
+
+        systemd_units.map(|units| Self {
+            systemd_units: units,
+        })
+    }
+}
+
+#[derive(Clone)]
+pub struct SystemDUnit {
+    name: String,
+    sections: HashMap<String, HashMap<String, String>>,
+    environment: HashMap<String, String>,
+}
+
+impl SystemDUnit {
+    /// Create a new unit which inherits all common elements from ['common_properties'] and parses
+    /// everything else from the ['container']
+
+    pub fn new(
+        common_properties: &SystemDUnit,
+        container: &Container,
+        pod_state: &PodState,
+    ) -> Result<Self, StackableError> {
+        let mut unit = common_properties.clone();
+        unit.name = String::from(container.name());
+
+        unit.add_property("Unit", "Description", &unit.name.clone());
+
+        unit.add_property(
+            "Service",
+            "ExecStart",
+            &SystemDUnit::get_command(container, pod_state)?,
+        );
+
+        let env_vars = SystemDUnit::get_environment(container, pod_state)?;
+
+        for (name, value) in env_vars {
+            unit.add_env_var(&name, &value);
+        }
+
+        unit.add_property("Service", "StandardOutput", "journal");
+        unit.add_property("Service", "StandardError", "journal");
+        unit.add_property("Install", "WantedBy", "multi-user.target");
+
+        Ok(unit)
+    }
+
+    pub fn new_from_pod(pod: &Pod) -> Self {
+        let mut unit = SystemDUnit {
+            name: pod.name().to_string(),
+            sections: Default::default(),
+            environment: Default::default(),
+        };
+
+        let restart_policy = match &pod.as_kube_pod().spec {
+            Some(spec) => spec.restart_policy.as_deref().unwrap_or("Never"),
+            None => "Never",
+        };
+
+        unit.add_property(
+            "Service",
+            "Restart",
+            RESTART_POLICY_MAP.get(restart_policy).unwrap(),
+        );
+        unit
+    }
+
+    // Add a key=value entry to the specified section
+    fn add_property(&mut self, section: &'static str, key: &str, value: &str) {
+        let section = self
+            .sections
+            .entry(String::from(section))
+            .or_insert_with(HashMap::new);
+        section.insert(String::from(key), String::from(value));
+    }
+
+    fn add_env_var(&mut self, name: &str, value: &str) {
+        self.environment
+            .insert(String::from(name), String::from(value));
+    }
+
+    pub fn get_unit_file_content(&self) -> String {
+        let mut unit_file_content = String::new();
+
+        // Iterate over all sections and write out its header and content
+        for (section, entries) in &self.sections {
+            unit_file_content.push_str(&format!("[{}]\n", section));
+            for (key, value) in entries {
+                unit_file_content.push_str(&format!("{}={}\n", key, value));
+            }
+            if section.eq("Service") {
```

Comment:
```suggestion
            if section.eq =="Service" {
```
seems more "normal" to me but I'm fine either way